app/views/shared/_assignments_dropdown_menu.html.erb

```diff
@@ -29,8 +29,12 @@
 <li class='section'><%= Assignment.model_name.human.pluralize %></li>
 
 <% if @current_user.student? %>
-  <% assignments = Assignment.where(is_hidden: false) %>
-  <% grade_entry_forms = GradeEntryForm.where(is_hidden: false) %>
+  <% assignments = Assignment.joins(:section_due_dates).where(is_hidden: false,
+                                                          'section_due_dates.section_id': @current_user.section,
```

Comment:
I'm surprised the linter didn't complain about this one...

You should align the arguments to the where method (like you did on lines 35-37)

Reply:
What will happen here when `@current_user.section` is `nil` or there isn't a `section_due_date` associated to the given section/assignment? (Hint: `joins` does an inner join by default)